tests/observability/metrics/utils.py

```diff
@@ -1283,6 +1285,127 @@ def get_metric_labels_non_empty_value(prometheus: Prometheus, metric_name: str)
     return {}
 
 
+def get_pod_memory_stats(admin_client: DynamicClient, hco_namespace: str, pod_prefix: str) -> float:
+    return float(
+        bitmath.Byte(
+            float(
+                get_pod_by_name_prefix(
+                    dyn_client=admin_client,
+                    pod_prefix=pod_prefix,
+                    namespace=hco_namespace,
+                ).execute(command=RSS_MEMORY_COMMAND)
+            )
+        )
+    )
+
+
+def get_highest_memory_usage_virt_api_pod_tuple(hco_namespace: str) -> tuple[str, int]:
+    """
+    This function returns pod name and memory value tuple of virt-api pod with the highest memory usage.
+    """
+    virt_api_with_highest_memory_usage = (
+        run_command(
+            command=shlex.split(
+                f"oc adm top pod -n {hco_namespace} --sort-by memory --no-headers -l kubevirt.io=virt-api"
```

Comment:
just proposal. if you add ` | head -n 1` here - you will have only one pod in return, so don't need to `.strip().split("\n")[1:]`

Reply:
what the error? here the output example:

```
$ oc adm top pod -n openshift-cnv --sort-by memory --no-headers -l kubevirt.io=virt-api | head -n 1virt-api-8597f69966-b27tn   2m    293Mi   
```